game/inventory/Inventory.java

```diff
@@ -84,4 +84,292 @@ public static int getCount(Filter<RSItem> filter) {
+                }
+            }
+
+            return itemList.toArray(new RSItem[itemList.size()]);
+        }
+
+        return new RSItem[0];
+    }
+
+    /**
+     * Gets items in lowest containing square of items of given width and height.
+     *
+     * @param width Width of square.
+     * @param items Items to get those contained within rectangle from.
+     * @return Items contained in lowest containing rectangle of items with given width/height in inventory.
+     */
+    public static RSItem[] getItemsInLowestSquare(int width, final RSItem... items) {
+
+        return getItemsInSquare(getLowestContainingRow(items), getLowestContainingColumn(items), width, items);
+    }
+
+    /**
+     * Gets given items contained within given square of items in inventory
+     *
+     * @param baseRow Start row to get items from
+     * @param baseCol Start column to get items from
+     * @param width   Width of area in number of items to get from.
+     * @param items   Items to get those contained within rectangle from.
+     * @return Items contained in rectangle in inventory.
+     */
+    public static RSItem[] getItemsInSquare(final int baseRow, final int baseCol, final int width, final RSItem... items) {
+
+        return getItemsInRectangle(baseRow, baseCol, width, width, items);
+    }
+
+    /**
+     * Gets the items that surround the given item.
+     * Uses the actual given item not just generically make sure its in the inventory.
+     *
+     * @param rsItem           item to get surrounding items of.
+     * @param surroundingWidth Width in number of items of the surrounding items.
+     *                         For example 1 to get a 1 item thick set of surrounding items
+     *                         like
+     *                         ***
+     *                         *i*
+     *                         ***
+     *                         or 2 to get a 2 thick set of surrounding items
+     *                         like
+     *                         *****
+     *                         *****
+     *                         **i**
+     *                         *****
+     *                         *****
+     *                         and so on.
+     * @return the items surrounding the given item in the inventory.
+     */
+    public static RSItem[] getSurroundingItems(final RSItem rsItem, int surroundingWidth) {
+
+        if (rsItem != null && hasItem(rsItem)) {
+
+            int row = getItemRowUnsafe(rsItem) - surroundingWidth;
+            int col = getItemColumnUnsafe(rsItem) - surroundingWidth;
+            surroundingWidth = (surroundingWidth * 2) + 1;
+            int surroundingHeight = surroundingWidth;
+
+            if (row < 0) {
+                surroundingHeight += row;
+                row = 0;
+            }
+            if (col < 0) {
+                surroundingWidth += col;
+                col = 0;
+            }
+
+            List<RSItem> itemList = new ArrayList<>();
```

Comment:
Do you need the extra list here? Cant you simply do 
`List<RSItem> itemList =Arrays.asList(...);`
instead?